triplea-game/triplea
# Upgrade to Java11 [Rawr!!] & Static Bundle Java11 with Installers

## Overview
- Sets project to build with Java 11
- Drops Java8 build from Travis
- Bundles Java11 with the install4j installers. Increases installer
  size by about 40MB to 60MB.
- Install4j JRE bundles set to be 'static'. This is an install4j config that
  instructs install4j to not share JREs between already installed install4j apps.
  It is not clear if this will be respected for the 'same' app when upgrading.

## Functional Changes
- Should be none. The intall4j config change is notable and of course the bundled JRE should mean that users will no longer be hit with "old java version" problems
- Drops win32 installer
- Installer size increases

## Manual Testing Performed
All seemed good:
- ran a gradle release locally and verified the installers were created
- compiled project in IDEA
- launched:
  - headless game runner
  - lobby
  - bot
  - http server
- ran a gradle check

I did *not* try out the mac or windows installers that were created. IMO we can go forward and take advantage of TripleA crowdsourcing (going oldschool) to do that testing and if we see any problems that we can't quickly resolve, revert is just a few button clicks away.

## Screen Shots
Note the config settings and that 'install as shared JRE' is not selected:
![Screenshot from 2019-06-06 23-01-18](https://user-images.githubusercontent.com/12397753/59084182-09685e80-88af-11e9-8862-1f9ce9e70495.png)


## Additional Review Notes
- this recent issue is related: https://github.com/triplea-game/triplea/issues/4874 and raises topic of us committing to Java11
- Java8 support is getting difficult, this branch: https://github.com/triplea-game/triplea/pull/4872 already needs us to drop Java8. Between the user problems we see with old Java 8 version, the problems in PR, and Java 8 getting pretty old, it seems we've hit a bit of a critical mass IMO to migrate.
